js/views/modals/listingDetail/Listing.js

```diff
@@ -307,6 +309,32 @@ export default class extends BaseModal {
     });
   }
 
+  startPurchase() {
+    const selectedVariants = [];
+    this.variantSelects.each((i, select) => {
+      const variant = {};
+      variant.name = $(select).attr('name');
+      variant.value = $(select).val();
+      selectedVariants.push(variant);
+    });
+
+    if (this.purchaseModal) {
```

Comment:
Since we're removing it on close, I don't think this case will ever be hit... but up to you if you want to leave it in.